.github/actions/build-src/action.yaml

```diff
@@ -9,6 +9,10 @@ inputs:
     description: 'build MDA docs'
     required: true
     default: false
+  isolation:
```

Comment:
We're adding an option to trigger the pip --no-build-isolation flag. This will allow us to avoid `pip` trying to have its own isolated build & overidding already installed dependencies.

Reply:
It took me a couple readings of this comment and the pip documentation to understand (if I have this correctly) that:
- default (i.e. isolate) installs new build-time dependencies that can be separate from runtime dependencies or previously installed packages
- `--no-build-isolation` does not override previously installed packages (I think?) but makes use of what is currently installed, which is why we added this flag here to test different build package versions

If that's right, could you please add that to the description of this flag, so later maintainers understand the use of the flag in our CI?

Reply:
Honestly I'm not really 100% sure I know exactly how build isolation works, my take is that it creates an isolated environment with brand new everything to create wheels and then it gives you the wheels back in the world you first started in.

I'll try to add something sensible here.

Reply:
Ok I've added a longer explanation and justification for how build isolation works, why we need it, etc...

I'm going to go ahead with the merge, but if you have the time to review what I wrote here and call out anything that's unclear in an issue that'd be great.